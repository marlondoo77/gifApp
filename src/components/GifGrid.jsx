
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";


//reac ejecuta este componente cada vez que detecta un cambio
export const GifGrid = ({categoria}) => {

  
  
  const { images, isLoading } = useFetchGifs( categoria );



  return (
    <>

        <h3>{ categoria }</h3>
        {
          isLoading && (<h2>Cargando ...</h2>) 
        }
        
        <div className="card-grid">
          {
            images.map( (imagen) => (
             <GifItem 
              key={imagen.id} 
              { ...imagen }
             />
            ))
          }
        </div>
    
    </>
  )
}
