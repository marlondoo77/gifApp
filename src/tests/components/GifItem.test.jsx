import { GifItem } from "../../components/GifItem"
import { render, screen } from "@testing-library/react"

describe('pruebas en gifitem componente', () => { 

    const title = 'Yo soy el titulo';
    const url = 'https://soylaurl.com/';
    
    test('debe de hacer match con el snapshot', () => { 
        
        const {container} = render( <GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    
     
     })

     test('debe de mostrar la imagen con el url y el alt indicado', () => { 
        
        const {container} = render( <GifItem title={title} url={url}/>)
        //screen.debug();
        //console.log(screen.getByRole('img'))
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

      })


      test('debe de mostrar el titulo en el componente', () => { 
        
        const {container} = render( <GifItem title={title} url={url}/>)

        expect( screen.getByText(title)).toBeTruthy();

       })


 })