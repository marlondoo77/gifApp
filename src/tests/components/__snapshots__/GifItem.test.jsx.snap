// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gifitem componente debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Yo soy el titulo"
      src="https://soylaurl.com/"
    />
    <p>
      Yo soy el titulo
    </p>
  </div>
</div>
`;
